### ISSUE TITLE
Transaction data changed after NFT Approve Modal

### ISSUE THREAD

**Issue by jingxuan98** [MetaMask/metamask-mobile#6800]
**Describe the bug**
The transaction data changed for the tokenId of approve(operator Address, tokenId) of ERC721, after the NFT Approve modal popup

Can be seen in my screen recording below as before the NFT Approve popup the tokenId is "1d", while after the NFT Approval popup, the transaction data is changed to 29 which results an error of "Invalid Token Id"

**Screenshots**

https://github.com/MetaMask/metamask-mobile/assets/43902263/c229ffc4-8ee1-4698-879f-dc3ea44d1cce

Results in "Invalid Token Id" because of changing the "1d" (hex value of 29) to 29
![Screenshot 2023-07-13 at 5 37 52 PM](https://github.com/MetaMask/metamask-mobile/assets/43902263/0f990b65-9645-4d10-9a15-da2721a4589f)
![Screenshot 2023-07-13 at 5 39 01 PM](https://github.com/MetaMask/metamask-mobile/assets/43902263/919616d4-c409-41d0-9dae-5e9bb6e43ba2)

**To Reproduce**
_Steps to reproduce the behavior_
1. Initiate a approve() of erc721
```
        const contract = new ethers.Contract(nftAddress, NFTABI, ethereumSigner)
         await contract.approve(toAddress, tokenId)
```
2. Observe the transaction data for metamask
3. I got this when using walletConnect + rainbow kit, tried the polygonscan approve and it works fine

**Expected behavior**
Should not change the transaction data after NFT Approve popup in metamask mobile wallet

**Smartphone (please complete the following information):**
 - Device: HONOR 50
 - OS: Android 12
 - App Version = 7.2.0
 
-------------------------------------------------------------
_to be added after bug submission by internal support / PM_
**Severity**

- How critical is the impact of this bug on a user? 
- Add stats if available on % of customers impacted
- Is this visible to all users? 
- Is this tech debt?

**Comment by asdullahsiddique** [comment]
Same happens to me on Metamask mobile on iOS. Was working before updating to the latest version of Metamask.

**Comment by 0xdsae** [comment]
Same happens to me on Metamask mobile on iOS. Was working before updating to the latest version of Metamask.

**Comment by AlexeyAdoniev** [comment]
Same bug, has started after july update of MetaMask android app
Cannot approve, tokeId doesn't encode to hex in 'data' field, is send as decimal with bunch of zeros..

**Comment by hanzochang** [comment]
I'm encountering a similar issue. After inspecting the approve contract, I've noticed that the hexadecimal value seems to be converted into something resembling a decimal value. This problem is causing difficulties for me in providing services.

**Comment by Bing72** [comment]
me 2

**Comment by anaamolnar** [comment]
Hello, @jingxuan98. Thank you so much for your report! I will pass this on to the team.

**Comment by bschorchit** [comment]
Thank you for your report @jingxuan98! We've created a fix for this here https://github.com/MetaMask/metamask-mobile/pull/6833 and it'll be included in our next release v7.4. It will likely be rolling out to users on the week of August 1st.
